The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15082202433/job/42400299513) [(plain)](https://github.com/rust-lang/rust/commit/604f0e27436b4940399873d0c23e5b609a52e831/checks/42400299513/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 4.286      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.1/6.1 MB 25.8 MB/s eta 0:00:00
#12 4.330 Collecting distlib<1,>=0.3.7
#12 4.338   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.349      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 49.0 MB/s eta 0:00:00
#12 4.388 Collecting platformdirs<5,>=3.9.1
#12 4.395   Downloading platformdirs-4.3.8-py3-none-any.whl (18 kB)
#12 4.436 Collecting filelock<4,>=3.12.2
#12 4.444   Downloading filelock-3.18.0-py3-none-any.whl (16 kB)
#12 4.527 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.721 Successfully installed distlib-0.3.9 filelock-3.18.0 platformdirs-4.3.8 virtualenv-20.31.2
#12 4.721 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.8s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      141248 kB
DirectMap2M:     9295872 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2025-05-12/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
[TIMING] core::build_steps::tool::ToolBuild { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu, forced_compiler: false }, target: x86_64-unknown-linux-gnu, tool: "tidy", path: "src/tools/tidy", mode: ToolBootstrap, source_type: InTree, extra_features: [], allow_features: "", cargo_args: [], artifact_kind: Binary } -- 35.411
[TIMING] core::build_steps::tool::Tidy { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu, forced_compiler: false }, target: x86_64-unknown-linux-gnu } -- 0.000
fmt check
fmt: checked 6014 files
Diff in /checkout/library/std/tests/path.rs:1998:
     assert_eq!(path.file_name(), Some(OsStr::new("file.txt")));
 }
 
-
 // Test: Reserved device names (Windows)
 // This test ensures that reserved device names like "CON", "PRN", etc., are handled as normal paths on non-Windows platforms,
 // and as special cases on Windows (if applicable).
Build completed unsuccessfully in 0:01:15
  local time: Sat May 17 05:51:16 UTC 2025
  network time: Sat, 17 May 2025 05:51:16 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/tests/path.rs
+++ b/library/std/tests/path.rs
@@ -1998,61 +1998,6 @@ fn test_non_ascii_unicode() {
     assert_eq!(path.file_name(), Some(OsStr::new("file.txt")));
 }
 
-
-// Test: Reserved device names (Windows)
-// This test ensures that reserved device names like "CON", "PRN", etc., are handled as normal paths on non-Windows platforms,
-// and as special cases on Windows (if applicable).
-#[test]
-#[cfg(windows)]
-fn test_reserved_device_names() {
-    for &name in &["CON", "PRN", "AUX", "NUL", "COM1", "LPT1"] {
-        let path = Path::new(name);
-        assert_eq!(path.file_name(), Some(OsStr::new(name)));
-        assert_eq!(path.extension(), None);
-    }
-}
-
-// Test: Trailing dots/spaces (Windows)
-// This test checks how Path handles trailing dots or spaces, which are special on Windows.
-// On Unix, these should be treated as normal characters.
-#[test]
-#[cfg(windows)]
-fn test_trailing_dots_and_spaces() {
-    let path = Path::new("foo. ");
-    assert_eq!(path.file_stem(), Some(OsStr::new("foo")));
-    assert_eq!(path.extension(), Some(OsStr::new(" ")));
-    assert_eq!(path.file_name(), Some(OsStr::new("foo. ")));
-    assert_eq!(path.to_str(), Some("foo. "));
-    let path = Path::new("bar...");
-    assert_eq!(path.file_stem(), Some(OsStr::new("bar")));
-    assert_eq!(path.extension(), Some(OsStr::new("...")));
-    assert_eq!(path.file_name(), Some(OsStr::new("bar...")));
-    assert_eq!(path.to_str(), Some("bar..."));
-}
-
-// Test: Only extension (e.g., ".gitignore")
-// This test verifies that files with only an extension and no base name are handled correctly.
-// It checks that the extension is recognized and the file stem is None or empty as appropriate.
-#[test]
-fn test_only_extension() {
-    let path = Path::new(".ext");
-    assert_eq!(path.extension(), None);
-    assert_eq!(path.file_stem(), Some(OsStr::new(".ext")));
-    assert_eq!(path.file_name(), Some(OsStr::new(".ext")));
-}
-
-// Test: Long components
-// This test checks that Path can handle very long path components without truncation or error.
-// It ensures that the length of the component is preserved.
-#[test]
-fn test_long_component() {
-    let long = "a".repeat(300);
-    let path = Path::new(&long);
-    assert_eq!(path.file_name(), Some(OsStr::new(&long)));
-    assert_eq!(path.to_str(), Some(long.as_str()));
-    assert_eq!(path.iter().count(), 1);
-}
-
 // Test: Embedded newlines
 // This test verifies that newlines within path components are preserved and do not break path parsing.
 // It ensures that Path treats newlines as normal characters.

--- a/library/std/tests/path.rs
+++ b/library/std/tests/path.rs
@@ -1235,7 +1235,7 @@ pub fn test_push() {
         tp!("foo//", "bar", r"foo//bar");
         tp!(r"foo\\", "bar", r"foo\\bar");
         tp!("foo/.", "bar", r"foo/.\bar");
-        tp!("foo./.", "bar", r"foo././bar");
+        tp!("foo./.", "bar", r"foo./.\bar");
         tp!(r"foo\.", "bar", r"foo\.\bar");
         tp!(r"foo.\.", "bar", r"foo.\.\bar");
         tp!("foo", "", "foo\\");

